File: Loenn/entities/locked_intro_car.lua
local drawableSprite = require "structs.drawable_sprite"
local utils = require "utils"

local lockedIntroCar = {}

local bodyTexture = "scenery/car/body"
local wheelsTexture = "scenery/car/wheels"

lockedIntroCar.name = "vitellary/lockedintrocar"

lockedIntroCar.placements = {
    {
        name = "locked_intro_car",
        data = {}
    }
}

function lockedIntroCar.sprite(room, entity)
    local sprites = {}

    local bodySprite = drawableSprite.fromTexture(bodyTexture, entity)
    bodySprite:setJustification(0.5, 1.0)
    bodySprite.depth = 1

    local wheelSprite = drawableSprite.fromTexture(wheelsTexture, entity)
    wheelSprite:setJustification(0.5, 1.0)
    wheelSprite.depth = 3

    table.insert(sprites, bodySprite)
    table.insert(sprites, wheelSprite)

    return sprites
end

function lockedIntroCar.selection(room, entity)
    local bodySprite = drawableSprite.fromTexture(bodyTexture, entity)
    local wheelSprite = drawableSprite.fromTexture(wheelsTexture, entity)

    bodySprite:setJustification(0.5, 1.0)
    wheelSprite:setJustification(0.5, 1.0)

    return utils.rectangle(utils.coverRectangles({bodySprite:getRectangle(), wheelSprite:getRectangle()}))
end

return lockedIntroCar

File: Ahorn/entities/lockedIntroCar.jl
module LockedIntroCar

using ..Ahorn, Maple

@mapdef Entity "vitellary/lockedintrocar" IntroCar(x::Integer, y::Integer)

const placements = Ahorn.PlacementDict(
    "Locked Intro Car (Crystalline)" => Ahorn.EntityPlacement(
        IntroCar
    )
)

bodySprite = "scenery/car/body"
wheelsSprite = "scenery/car/wheels"

function Ahorn.selection(entity::IntroCar)
    x, y = Ahorn.position(entity)

    rectangles = Ahorn.Rectangle[
        Ahorn.getSpriteRectangle(bodySprite, x, y, jx=0.5, jy=1.0),
        Ahorn.getSpriteRectangle(wheelsSprite, x, y, jx=0.5, jy=1.0),
    ]

    return Ahorn.coverRectangles(rectangles)
end

function Ahorn.renderAbs(ctx::Ahorn.Cairo.CairoContext, entity::IntroCar, room::Maple.Room)
    x, y = Ahorn.position(entity)

    Ahorn.drawSprite(ctx, wheelsSprite, x, y, jx=0.5, jy=1.0)
    Ahorn.drawSprite(ctx, bodySprite, x, y, jx=0.5, jy=1.0)
end

end